stistools/ocrreject_exam.py

```diff
@@ -373,38 +390,48 @@ def stack_plot(stack_image, box_lower, box_upper, split_num, texpt, obs_id, prop
 
     if not interactive:
         # create matplotlib image
-        fig, (ax1, ax2, ax3) = plt.subplots(nrows=1, ncols=3, figsize=(9, 20*(9/41)),
-            gridspec_kw={'width_ratios': [1, 1, 0.05], 'height_ratios': [1]})
-
-        for axis in [ax1, ax2]:
-            axis.imshow(stack_image, interpolation='none', origin='lower',
-                extent=(0, stack_shape[1], 0, stack_shape[0]), cmap=cmap, norm=norm, aspect='auto')
-            axis.step(np.arange(len(box_upper)), box_upper, color='#222222', where='post', lw=0.7, alpha=0.7, ls='--')
-            axis.step(np.arange(len(box_lower)), box_lower, color='#222222', where='post', lw=0.7, alpha=0.7, ls='--')
-
-        ax1.set_title('Full image')
-
-        # If it is a large enough image, zoom the 2nd subplot around the extraction box region
-        if ((stack_shape[0] - max(box_upper)) > 20) and (min(box_lower) > 20):
-            ax2.set_ylim([(min(box_lower)-20),(max(box_upper)+20)])
-            ax2.set_title('zoomed to 20 pixels above/below extraction box')
-
-        # Otherwise just don't zoom in at all
-        else:
-            ax2.set_title('full image already 20 pixels above/below extraction box')
+        is_interactive_backend = matplotlib.get_backend().lower() not in {"agg", "pdf", "svg", "ps"}
```

Comment:
Since this code is repeated, I'd suggest creating a global variable at the top and using it in the two locations:

```python
NON_INTERACTIVE_BACKENDS = {"agg", "pdf", "svg", "ps"}
```

Reply:
@m-dallas -
I think this is the last change needed.